perun-wui-profile/src/main/java/cz/metacentrum/perun/wui/profile/pages/settings/altPasswords/AltPasswordsView.java

```diff
@@ -0,0 +1,146 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+
+public class AltPasswordsView extends ViewWithUiHandlers<AltPasswordsUiHandlers> implements AltPasswordsPresenter.MyView {
+
+	interface AltPasswordsViewUiBinder extends UiBinder<Widget, AltPasswordsView> {}
+
+	private PerunProfileTranslation translation = GWT.create(PerunProfileTranslation.class);
+
+	@UiField
+	TextBox passwordBox;
+
+	@UiField
+	PerunButton addAltPasswordButton;
+
+	@UiField
+	DataGrid<Map.Entry<String, String>> aPDataGrid;
+
+	@UiField
+	Div panelsDiv;
+
+	@UiField
+	Modal generatedPasswordModal;
+
+	@UiField
+	TextBox generatedPasswordBox;
+
+	@UiHandler({"addAltPasswordButton"})
+	public void onListAll(ClickEvent event) {
+		getUiHandlers().newAltPassoword(passwordBox.getText(), generatePassword(16));
+	}
+
+	@UiHandler("backButton")
+	public void cancelButtonAction(ClickEvent event) {
+		getUiHandlers().navigateBack();
+	}
+
+	@Inject
+	public AltPasswordsView(AltPasswordsViewUiBinder binder) {
+		initWidget(binder.createAndBindUi(this));
+		passwordBox.setWidth("25%");
+		addAltPasswordButton.setWidth("25%");
+		initTable(aPDataGrid);
+		generatedPasswordModal.addShownHandler(modalShownEvent -> generatedPasswordBox.setFocus(true));
+	}
+
+	@Override
+	public void onError(PerunException e) {	}
+
+	@Override
+	public void setData(Map<String, JSONValue> data) {
+		if (data.isEmpty()) {
+			((PerunLoader) aPDataGrid.getEmptyTableWidget()).onEmpty();
+			aPDataGrid.setRowCount(0);
+		} else {
+			Map<String, String> tableData = new HashMap<>();
+			for (Map.Entry<String, JSONValue> entry : data.entrySet()) {
+				tableData.put(entry.getKey(), entry.getValue().toString());
+			}
+			aPDataGrid.setRowData(new ArrayList<>(tableData.entrySet()));
+		}
+
+	}
+
+	@Override
+	public void onLoadingStart() {
+		((PerunLoader) aPDataGrid.getEmptyTableWidget()).onLoading(translation.loadingUserData());
+	}
+
+	private void initTable(DataGrid<Map.Entry<String, String>> dataGrid){
+
+		TextColumn<Map.Entry<String, String>> nameCol = new TextColumn<Map.Entry<String, String>>() {
+
+			@Override
+			public String getValue(Map.Entry<String, String> entry) {
+				return entry.getKey();
+			}
+		};
+
+		Column<Map.Entry<String, String>, String> deleteCol =
+				new Column<Map.Entry<String, String>, String>(new ButtonCell(ButtonType.DANGER)) {
+
+					@Override
+					public String getValue(Map.Entry<String, String> entry) {
+						return translation.delete();
+					}
+				};
+
+		deleteCol.setFieldUpdater((i, entry, s) -> getUiHandlers().deleteAltPassword(entry.getValue()));
+		PerunLoader pl = new PerunLoader();
+		pl.setEmptyMessage(translation.noPasswords());
+		pl.getElement().getStyle().setMarginTop(20, Style.Unit.PX);
+		dataGrid.setEmptyTableWidget(pl);
+		dataGrid.addColumn(nameCol, translation.description());
+		dataGrid.addColumn(deleteCol, "");
+		dataGrid.setColumnWidth(nameCol, "91%");
+		dataGrid.setColumnWidth(deleteCol, "9%");
```

Comment:
Please increase the column width. With Czech lang button is partially hidden. Maybe play with data grid settings. I'm unsure if it will be correct if not exact width is set.